### ISSUE TITLE
Reflect the errors of report generating

### ISSUE THREAD

**Issue by tromai** [oracle-samples/macaron#176]
**Description**
At the moment, all of our Reporter [here](https://github.com/oracle-samples/macaron/blob/b009eb0496d3dec1c39d5972b3c1d8622634e831/src/macaron/output_reporter/reporter.py) does not return anything even when errors happen while generating the reports.
Therefore, it's better for us to return the error accordingly so that the Analyzer could decide on the exit status code (https://github.com/oracle-samples/macaron/issues/175).

**Comment by behnazh-w** [comment]
> **Description** At the moment, all of our Reporter [here](https://github.com/oracle-samples/macaron/blob/b009eb0496d3dec1c39d5972b3c1d8622634e831/src/macaron/output_reporter/reporter.py) does not return anything even when errors happen while generating the reports. Therefore, it's better for us to return the error accordingly so that the Analyzer could decide on the exit status code (#175).

Do you have any specific scenario in mind? In most cases the report generator should not make the process exit.

**Comment by tromai** [comment]
> In most cases the report generator should not make the process exit.

Yes that is true. I was thinking about whether we should explicitly show the user that the reports could not be generated (of an expected internal error). For example, at the moment, if Macaron cannot generate the report, the return status code would still be 0 (no error) if the analysis succeed. I am thinking that the user might not aware of it and look into the old version of the reports or could not find the report anywhere.

**Comment by behnazh-w** [comment]
> I am thinking that the user might not aware of it and look into the old version of the reports or could not find the report anywhere.

We should remove old results before producing new results to avoid this. I'm not sure if we should return an error code, which would be considered as Macaron's analysis failure.
